### PR TITLE
Updates TextInput to use FormField

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Review by audroue — APPROVED** [review]

File: packages/text-input/src/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -93,11 +65,10 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
     }: TextInputProps,
     forwardRef: React.Ref<HTMLInputElement>,
   ) => {
-    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const { darkMode } = useDarkMode(darkModeProp);
     const isControlled = typeof controlledValue === 'string';
     const [uncontrolledValue, setValue] = useState('');
     const value = isControlled ? controlledValue : uncontrolledValue;
-    const id = useIdAllocator({ prefix: 'textinput', id: propsId });
```
Comment (spark33) [inline packages/text-input/src/TextInput/TextInput.tsx:100]: Would this change make this a breaking change? If users were dependent on the value of this id always being `textinput-0` for example?

File: packages/text-input/src/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -152,109 +123,37 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
...
-            {shouldRenderOptionalText && (
-              <div
-                className={cx(
-                  optionalTextBaseStyle,
-                  optionalTextThemeStyle[theme],
-                )}
-              >
-                <p>Optional</p>
-              </div>
-            )}
```
Comment (spark33) [inline packages/text-input/src/TextInput/TextInput.tsx:249]: It looks like the `optional` text styles flash for a little bit on initial render

File: packages/text-input/src/TextInput/TextInput.styles.ts
Comment (spark33) [inline packages/text-input/src/TextInput/TextInput.styles.ts:1]: Crazy we could just straight up delete this file 🤯

File: .changeset/orange-seahorses-fix.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/text-input': patch
+---
+
+Refactors component to use FormField under the hood. Also, per the latest design guidelines, we no longer surface an error message when an input is disabled.
```
Comment (spark33) [inline .changeset/orange-seahorses-fix.md:5]: Shouldn't an introduction of a net new dependency be considered major?

File: packages/text-input/src/TextInput/TextInput.tsx (hunk 2 above)
Comment (spark33) [reply packages/text-input/src/TextInput/TextInput.tsx:249]: Nvm - this happens on our current Storybook deployment too

File: packages/text-input/src/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -152,109 +123,37 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
...
+        description={description}
+        errorMessage={errorMessage}
+        state={state}
+        size={sizeVariant}
+        disabled={disabled}
+        baseFontSize={baseFontSize}
+        darkMode={darkMode}
+        className={className}
+        id={id}
+        optional={optional}
```
Comment (spark33) [inline packages/text-input/src/TextInput/TextInput.tsx:138]: Maybe this was discussed while I was out, but is baking in `optional` into FormField the best choice? Isn't this `optional` text only in TextInput?

**Review by spark33 — COMMENTED** [review]
Code and the live example both look great. I just have thoughts on the API of FormField and whether this is just a patch update.

**Review by spark33 — APPROVED** [review]

**Pull request merged**